unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/MyAccountHUD/AdditionalInfoOptionsModel.cs

```diff
@@ -23,8 +24,50 @@ public class Option
             public InputType InputType { get; set; }
             [CanBeNull] public string[] Values { get; set; }
             [CanBeNull] public string DateFormat { get; set; }
+
+            protected bool Equals(Option other) =>
+                IsAvailable == other.IsAvailable && Name == other.Name && InputType == other.InputType && Equals(Values, other.Values) && DateFormat == other.DateFormat;
+
+            public override bool Equals(object obj)
+            {
+                if (ReferenceEquals(null, obj)) return false;
+                if (ReferenceEquals(this, obj)) return true;
+                if (obj.GetType() != this.GetType()) return false;
+                return Equals((Option)obj);
+            }
+
+            public override int GetHashCode() =>
+                HashCode.Combine(IsAvailable, Name, (int)InputType, Values, DateFormat);
         }
 
         public Dictionary<string, Option> Options { get; set; }
+
+        protected bool Equals(AdditionalInfoOptionsModel other)
```

Comment:
Theoretically, it can be very expensive considering that iterating over `dictionary` is not very efficient (in comparison to `array`). How many elements does it contain and how often do we call it?

Reply:
The point of this `Equals` was to avoid updating a whole UI when the data model changed and this object had no changes, The UI implementation Destroys and Instantiates a bunch of game objects every time this `Options` dictionary changes.

So in terms of efficiency we are winning a lot.

Reply:
This fixes the hiccup when triggering an Emote 😆